File: PTA/advanced_level/1149.py
n, m = [int(i) for i in input().split()]
mp = {}
for i in range(n):
	a, b = [int(j) for j in input().split()]
	if a not in mp: mp[a] = []
	if b not in mp: mp[b] = []
	mp[a].append(i)
	mp[b].append(i)
for i in range(m):
	li = [int(j) for j in input().split()]
	del li[0]
	bad = [0 for j in range(n)]
	good = True
	for j in li:
		if j not in mp: continue
		for a in mp[j]:
			bad[a] += 1
			if bad[a] == 2:
				good = False
				break
	if good: print("Yes")
	else: print("No")
		

File: PTA/advanced_level/1149.cpp
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <queue>
using namespace std;
struct node{
    int next;
    int bad;
};
node edge[20005];
int head[100005],n,m,from,to,cnt,bad_cnt,num,temp;
int bad[10005];
//上面的3行，所有的值都会被默认赋值为0
void add(){//建立线性查询表
    edge[++cnt].next=head[from];
    edge[cnt].bad=bad_cnt;
    head[from]=cnt;
    
    edge[++cnt].next=head[to];
    edge[cnt].bad=bad_cnt++;
    head[to]=cnt;
    
}
int main(){
    scanf("%d %d\n",&n,&m);//第一行输入
    for(int i=0;i<n;i++){//不可互存物品输入
        scanf("%d %d\n",&from,&to);
        add();
    }
    for(int c=0;c<m;c++){//m次迭代
        scanf("%d\n",&num);
        fill(bad,bad+bad_cnt,0);//将bad [0,bad_cnt)这个范围内的值都赋值为0
        bool good=true;
        for(int i=0;i<num;i++){
            scanf("%d",&temp);
            getchar();
            for(int j=head[temp];j;j=edge[j].next){
                bad[edge[j].bad]++;
                if(bad[edge[j].bad]==2){
                    good=false;
                    break;
                }
            }
        }
        if(good) printf("Yes\n");
        else printf("No\n");
    }
	return 0;
}